src/Databases/DatabaseOrdinary.cpp

```diff
@@ -215,8 +216,11 @@ LoadTaskPtr DatabaseOrdinary::startupTableAsync(
                 logAboutProgress(log, ++tables_started, total_tables_to_startup, startup_watch);
             }
             else
-                throw Exception(ErrorCodes::LOGICAL_ERROR, "Table {}.{} doesn't exist during startup",
+            {
+                // This might happen if synchronous metadata loading failed and server is going to shutdown.
+                throw Exception(ErrorCodes::UNKNOWN_TABLE, "Table {}.{} doesn't exist during startup",
```

Comment:
Since we wait for the current jobs, it cannot happen anymore, right?

Reply:
Indeed. At least I'm not aware of other reasons for IStorage::shutdown() to be called before IStorage::startup(). Okay, lets make it back to LOGICAL_ERROR, so we can see if it breaks.